File: regions/system/library/dart/websocket/lib/src/ws_config.dart
class WsConfig {
  final String url;
  final bool reconnect;
  final int maxReconnectAttempts;
  final Duration reconnectDelay;
  final Duration? pingInterval;

  const WsConfig({
    required this.url,
    this.reconnect = true,
    this.maxReconnectAttempts = 5,
    this.reconnectDelay = const Duration(seconds: 1),
    this.pingInterval,
  });
}

File: regions/system/library/swift/websocket/Sources/K1s0WebSocket/WsConfig.swift
import Foundation

public struct WsConfig: Sendable {
    public let url: String
    public let reconnect: Bool
    public let maxReconnectAttempts: Int
    public let reconnectDelay: Duration
    public let pingInterval: Duration?

    public init(
        url: String = "ws://localhost",
        reconnect: Bool = true,
        maxReconnectAttempts: Int = 5,
        reconnectDelay: Duration = .seconds(1),
        pingInterval: Duration? = nil
    ) {
        self.url = url
        self.reconnect = reconnect
        self.maxReconnectAttempts = maxReconnectAttempts
        self.reconnectDelay = reconnectDelay
        self.pingInterval = pingInterval
    }

    public static let defaults = WsConfig()
}
